tests/components/tplink/__init__.py

```diff
@@ -95,6 +102,69 @@
 }
 
 
+def _load_feature_fixtures():
+    fixtures = load_json_value_fixture("features.json", DOMAIN)
+    for fixture in fixtures.values():
+        if isinstance(fixture["value"], str):
+            try:
+                time = datetime.strptime(fixture["value"], "%Y-%m-%d %H:%M:%S.%f%z")
+                fixture["value"] = time
+            except ValueError:
+                pass
+    return fixtures
+
+
+FEATURES_FIXTURE = _load_feature_fixtures()
+
+
+async def setup_platform_for_device(
+    hass: HomeAssistant, config_entry: ConfigEntry, platform: Platform, device: Device
+):
+    """Set up a single tplink platform with a device."""
+    config_entry.add_to_hass(hass)
+
+    with (
+        patch("homeassistant.components.tplink.PLATFORMS", [platform]),
+        _patch_discovery(device=device),
+        _patch_connect(device=device),
+    ):
+        await async_setup_component(hass, DOMAIN, {DOMAIN: {}})
+        await hass.async_block_till_done(wait_background_tasks=True)
+
+
+async def snapshot_platform(
```

Comment:
There is a `snapshot_platform` in [tests/common.py](https://github.com/home-assistant/core/blob/f1ddf80dff8644b4aab33cf966896faf0673bb47/tests/common.py#L1757) that was my original inspiration for this.  However it does not snapshot the `device_registry` which is quite important for us with our child devices.

I could certainly look to raise a PR to mainline that as an optional parameter however I should note that I experienced intermittent issues snapshotting our platforms when multiple sensors were defaulting to using default `device_class` names. I think it was because the order of entity creation was not fully deterministic and entities would be given `_timestamp`, `_timestamp_2` entity ids in a different order. Mainlining a check for deterministic entity_id naming could be more complex and depend on the integration implementation, although maybe the function could accept a callback...